src/vs/editor/common/model/textModel.ts

```diff
@@ -429,9 +429,8 @@ export class TextModel extends Disposable implements model.ITextModel, IDecorati
 
 	public setValue(value: string | model.ITextSnapshot): void {
 		this._assertNotDisposed();
-		if (value === null) {
-			// There's nothing to do
-			return;
+		if (value === null || value === undefined) {
+			value = '';
```

Comment:
The `setValue` function should throw an error (`IllegalArgument`) when invoked with `null` or `undefined`. The caller is violating the contract defined in the argument's type: `string | model.ITextSnapshot`, so the caller of `setValue(undefined)` has a bug.

Reply:
Thank you for the clarification.